diff --git a/node_modules/@moodlehq/cordova-plugin-qrscanner/src/ios/QRScanner.swift b/node_modules/@moodlehq/cordova-plugin-qrscanner/src/ios/QRScanner.swift
index c4033c838..de02f8148 100644
--- a/node_modules/@moodlehq/cordova-plugin-qrscanner/src/ios/QRScanner.swift
+++ b/node_modules/@moodlehq/cordova-plugin-qrscanner/src/ios/QRScanner.swift
@@ -238,10 +238,12 @@ class QRScanner : CDVPlugin, AVCaptureMetadataOutputObjectsDelegate {
             return
         }
         let found = metadataObjects[0] as! AVMetadataMachineReadableCodeObject
-        if found.type == AVMetadataObject.ObjectType.qr && found.stringValue != nil {
+        if found.type == AVMetadataObject.ObjectType.qr,
+            let value = found.stringValue,
+            let scanCommand = nextScanningCommand {
             scanning = false
-            let pluginResult = CDVPluginResult(status: CDVCommandStatus_OK, messageAs: found.stringValue)
-            commandDelegate!.send(pluginResult, callbackId: nextScanningCommand?.callbackId!)
+            let pluginResult = CDVPluginResult(status: CDVCommandStatus_OK, messageAs: value)
+            commandDelegate!.send(pluginResult, callbackId: scanCommand.callbackId)
             nextScanningCommand = nil
         }
     }
@@ -467,8 +469,7 @@ class QRScanner : CDVPlugin, AVCaptureMetadataOutputObjectsDelegate {
     }
 
     @objc func openSettings(_ command: CDVInvokedUrlCommand) {
-        if #available(iOS 10.0, *) {
-            guard let settingsUrl = URL(string: UIApplicationOpenSettingsURLString) else {
+        guard let settingsUrl = URL(string: UIApplication.openSettingsURLString) else {
             return
         }
         if UIApplication.shared.canOpenURL(settingsUrl) {
@@ -477,15 +478,6 @@ class QRScanner : CDVPlugin, AVCaptureMetadataOutputObjectsDelegate {
             })
         } else {
             self.sendErrorCode(command: command, error: QRScannerError.open_settings_unavailable)
-            }
-        } else {
-            // pre iOS 10.0
-            if #available(iOS 8.0, *) {
-                UIApplication.shared.openURL(NSURL(string: UIApplicationOpenSettingsURLString)! as URL)
-                self.getStatus(command)
-            } else {
-                self.sendErrorCode(command: command, error: QRScannerError.open_settings_unavailable)
-            }
         }
     }
 }
