diff --git a/node_modules/@moodlehq/cordova-plugin-file-transfer/src/ios/CDVFileTransfer.m b/node_modules/@moodlehq/cordova-plugin-file-transfer/src/ios/CDVFileTransfer.m
index 4134c02..c485861 100644
--- a/node_modules/@moodlehq/cordova-plugin-file-transfer/src/ios/CDVFileTransfer.m
+++ b/node_modules/@moodlehq/cordova-plugin-file-transfer/src/ios/CDVFileTransfer.m
@@ -19,6 +19,7 @@ Licensed to the Apache Software Foundation (ASF) under one
 
 #import <Cordova/CDV.h>
 #import "CDVFileTransfer.h"
+#import "CDVFile.h"
 #import "CDVLocalFilesystem.h"
 
 #import <AssetsLibrary/ALAsset.h>
@@ -324,11 +325,12 @@ - (void)fileDataForUploadCommand:(CDVInvokedUrlCommand*)command
     }
 
     CDVFilesystemURL *sourceURL = [CDVFilesystemURL fileSystemURLWithString:source];
+    CDVFile *filePlugin = [self.commandDelegate getCommandInstance:@"File"];
     NSObject<CDVFileSystem> *fs;
     if (sourceURL) {
         // Try to get a CDVFileSystem which will handle this file.
         // This requires talking to the current CDVFile plugin.
-        fs = [[self.commandDelegate getCommandInstance:@"File"] filesystemForURL:sourceURL];
+        fs = [filePlugin filesystemForURL:sourceURL];
     }
     if (fs) {
         __weak CDVFileTransfer* weakSelf = self;
@@ -430,7 +432,8 @@ - (void)download:(CDVInvokedUrlCommand*)command
          * Check here to see if it looks like the user passed in a raw filesystem path. (Perhaps they had the path saved, and were previously using it with the old version of File). If so, normalize it by removing empty path segments, and check with File to see if any of the installed filesystems will handle it. If so, then we will end up with a filesystem url to use for the remainder of this operation.
          */
         target = [target stringByReplacingOccurrencesOfString:@"//" withString:@"/"];
-        targetURL = [[self.commandDelegate getCommandInstance:@"File"] fileSystemURLforLocalPath:target].url;
+        CDVFile *filePlugin = [self.commandDelegate getCommandInstance:@"File"];
+        targetURL = [filePlugin fileSystemURLforLocalPath:target].url;
     } else {
         targetURL = [NSURL URLWithString:target];
 
